services/src/main/java/org/keycloak/services/clientpolicy/ClientUpdateContext.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.services.clientpolicy;
+
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.JsonWebToken;
+import org.keycloak.representations.idm.ClientRepresentation;
+
+/**
+ * Represents the context in the client registration/update by Dynamic Client Registration or Admin REST API.
+ */
+public interface ClientUpdateContext extends ClientPolicyContext {
```

Comment:
Does this interface need to have default methods? I see that most of the implementations need to override all the methods (with the exception of getClientToBeUpdated in case of "Register" classes). IMO it will be better if this interface don't have default methods - maybe with the exception of "getClientToBeUpdated" method.

Reply:
Yes, that's right. I'll only implement default method for `getClientToBeUpdated()` and remove default methods for others.

Reply:
+1, Thanks!